Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -624,18 +633,22 @@ extension ActivityStreamPanel: DataObserverDelegate {
 
         let removeTopSiteAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
             self.telemetry.reportEvent(.Remove, source: pingSource, position: index)
-            self.hideURLFromTopSites(site.tileURL)
+            self.hideURLFromTopSites(site)
         })
 
         let dismissHighlightAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
             self.telemetry.reportEvent(.Dismiss, source: pingSource, position: index)
             self.hideFromHighlights(site)
         })
 
+        let pinTopSite = ActionOverlayTableViewAction(title: "Pin TopSite", iconString: "action_share", handler: { action in
```

Comment:
You can still pin a site again if its pinned. Thats how you can move a pinned site to the front of the list if you want to change the order. 
If you want to remove a pinned site its the same as just removing a regular topsite.